src/gazebo_gst_camera_plugin.cpp

```diff
@@ -97,10 +97,38 @@ void GstCameraPlugin::startGstThread() {
   GstElement* dataSrc = gst_element_factory_make("appsrc", "AppSrc");
   GstElement* testSrc = gst_element_factory_make("videotestsrc", "FileSrc");
   GstElement* conv  = gst_element_factory_make("videoconvert", "Convert");
-  GstElement* encoder = gst_element_factory_make("x264enc", "AvcEncoder");
+
+  GstElement* encoder = gst_element_factory_make("nvh264enc", "AvcEncoder");
```

Comment:
This will always use `nvh264enc` whenever it is available. I am not sure if this is the expected behavior?

Reply:
The use case is to have SITL not just run for CI, but also to have a full stack of a drone solution available within a simulation (mqtt_bridge, rtmp streaming, ...) in order to decouple programming of drone software and groundcontrol software.

Reply:
@dayjaby I think if we make this as the default untested, it will fail for many people. Can we make this optional?

Reply:
@Jaeyoung-Lim It's only the default if it's available. I don't see a case where this is failing for anyone. Or am I mistaken?

```
if (!encoder) {
    encoder = gst_element_factory_make("x264enc", "AvcEncoder");
```
falls back to x264enc if nvh264enc is not available. (Sorry I forgot to push the `encoder = ...` here!)

Reply:
@dayjaby I agree, but I am just not comfortable on having a case that is not being tested. I agree falling back as default is a better implementation, but we need to test it if we want to have this behavior, or explicitly disable it as default